store/storage/sqlite/db.go

```diff
@@ -174,14 +189,44 @@ func (db *Database) ApplyChangeset(version uint64, cs *store.Changeset) error {
 	return b.Write()
 }
 
+// Prune removes all versions of all keys that are <= the given version. It keeps
+// the latest (non-tombstoned) version of each key/value tuple to handle queries
+// above the prune version. This is analogous to RocksDB full_history_ts_low.
+//
+// We perform the prune by deleting all versions of a key, excluding reserved keys,
+// that are <= the given version, except for the latest version of the key.
 func (db *Database) Prune(version uint64) error {
-	stmt := "DELETE FROM state_storage WHERE version <= ? AND store_key != ?;"
+	tx, err := db.storage.Begin()
+	if err != nil {
+		return fmt.Errorf("failed to create SQL transaction: %w", err)
+	}
 
-	_, err := db.storage.Exec(stmt, version, reservedStoreKey)
+	pruneStmt := `DELETE FROM state_storage
+	WHERE version < (
+		SELECT max(version) FROM state_storage t2 WHERE
+		t2.store_key = state_storage.store_key AND
+		t2.key = state_storage.key AND
+		t2.version <= ?
+	) AND store_key != ?;
```

Comment:
new table (max_version, key, store_key) could be helpful? this query seems like very heavy